code/src/java/pcgen/io/PCGIOHandler.java

```diff
@@ -633,7 +615,7 @@ private SourceSelectionFacade internalReadSources(InputStream in)
 		// Verify it is ver2
 		boolean isPCGVersion2 = isPCGCersion2(lines);
 
-		final String[] pcgLines = lines.toArray(new String[lines.size()]);
+		final String[] pcgLines = lines.toArray(new String[0]);
```

Comment:
Why lose the known length information and force 2 array creations rather than one?

Reply:
In pre-Java-6 versions using a presized array was better, as the reflection call  to properly size thw array slow. However since Kava 6 this call was intrinsified, making the performance of the empty array version the same and sometimes even better